### PR TITLE
[CM H1] Minor update on COMPILE.md

### PR DESCRIPTION
**Pull request by a-kraschitzer**
## Basics

These points need to be fulfilled for every PR:

- [X] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [X] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [X] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by a-kraschitzer** [comment]
Going over the INSTALL.md and COMPILE.md multiple times, I didn't really find to much where I could improve.
I was thinking of splitting up COMPILE.md into COMPILE_Ubuntu.md, COMPILE_CentOS.md because it is quite hard to find the relevant bits in the file.

File: doc/COMPILE.md [hunk 1]
```diff
@@ -82,17 +82,22 @@ apt-get install pdflatex \
         texlive-math-extra
 ```
 
-For the plugins, please refer to the README.md of the respective plugin.
-For example, for CentOS:
+For the plugins, please refer to the [README.md](https://www.libelektra.org/plugins/readme) of the respective plugin.
 
+A small subset of plugins to get you started:
+for CentOS:
```
Comment (mpranj) [inline doc/COMPILE.md:88]: This looks a bit odd when rendered, as there is no newline.

Maybe use a list to indent "for CentOS" and other sections (or use some other structure).
e.g.:
```suggestion
A small subset of plugins to get you started:
* for CentOS:
```

File: doc/COMPILE.md [hunk 2]
```diff
@@ -82,17 +82,22 @@ apt-get install pdflatex \
...
 ```sh
 sudo yum install -y boost-devel libdb-devel GConf2-devel libxml2-devel yajl-devel   \
 libcurl-devel augeas-devel libgit2-devel lua-devel swig python34-devel python-devel \
 java-1.8.0-openjdk-devel jna ruby-devel byacc
 ```
 
-For the Debian package, please refer to debian/control (in the debian
-branch).
+for Ubuntu:
```
Comment (mpranj) [inline doc/COMPILE.md:95]: list example continued

```suggestion
* for Ubuntu:
```

File: doc/COMPILE.md [hunk 3]
```diff
@@ -220,7 +225,7 @@ To add also experimental plugins, you can use:
 -DPLUGINS=ALL
 ```
 
-> Note that plugins are excluded automatically if dependences are not satisfied.
+> Note that plugins are only build if their dependencies are satisfied.
```
Comment (mpranj) [inline doc/COMPILE.md:228]:
```suggestion
> Note that plugins are only built if their dependencies are satisfied.
```

**Review by mpranj — COMMENTED** [review]
Thank you so much, looks good to me! I only added some minor suggestions.

Please rebase from master to resolve the merge conflict.

**Comment by mpranj** [comment]
> I was thinking of splitting up COMPILE.md into COMPILE_Ubuntu.md, COMPILE_CentOS.md 

Not needed _imho_. The quick introduction is right at the beginning of the file, followed by some deb/rpm specific stuff. The bulk of the file are just cmake options which apply to all platforms.

File: doc/COMPILE.md [hunk 4]
```diff
@@ -82,17 +82,22 @@ apt-get install pdflatex \
...
 java-1.8.0-openjdk-devel jna ruby-devel byacc
 ```
 
-For the Debian package, please refer to debian/control (in the debian
-branch).
+for Ubuntu:
+```sh
+sudo apt install -y libboost-dev libxerces-c-dev libxml2-dev libyajl-dev \
+libcurl4-gnutls-dev libaugeas-dev git git-buildpackage dh-lua liblua5.2-dev \
+dh-python python3-all python3-dev default-jdk libjna-java ruby-dev flex bison
```
Comment (mpranj) [inline doc/COMPILE.md:99]: trailing whitespace

```suggestion
dh-python python3-all python3-dev default-jdk libjna-java ruby-dev flex bison
```

File: doc/news/_preparation_next_release.md [hunk 5]
```diff
@@ -204,6 +204,7 @@ _(@tucek)_
 - <<TODO>>
 - <<TODO>>
 - <<TODO>>
+- Work on COMPILE.md and INSTALL.md to help with understanding _(@a-kraschitzer)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:207]: Thank you for the PR! Please fix the typos found by @mpranj (@mpranj thank you for the review!). Please also add links in the release notes (_preparation_next_release.md ).

**Comment by markus2330** [comment]
@a-kraschitzer can you please rebase? (Needed to due release.)

File: doc/COMPILE.md [hunk 6]
```diff
@@ -82,17 +82,25 @@ apt-get install pdflatex \
         texlive-math-extra
 ```
 
-For the plugins, please refer to the README.md of the respective plugin.
-For example, for CentOS:
+For the plugins, please refer to the [README.md](https://www.libelektra.org/plugins/readme) of the respective plugin.
+
+A small subset of plugins to get you started:
```
Comment (mpranj) [inline doc/COMPILE.md:87]:
```suggestion
For dependencies of plugins, please refer to the [README.md](https://www.libelektra.org/plugins/readme) of the respective plugin.

A small subset of build dependencies to get you started:
```

**Comment by mpranj** [comment]
Thank you so much!

**Pull request merged**